server/src/lib.rs

```diff
@@ -83,11 +83,18 @@ fn run() -> Result<ServerWrapper> {
         Err(e)
     })?;
 
-    let core_thread =
-        core::spawn_thread("AudioIPC Server RPC", move || Ok(()), || {}).or_else(|e| {
-            debug!("Failed to cubeb audio core event loop thread: {:?}", e);
-            Err(e)
-        })?;
+    let core_thread = core::spawn_thread(
+        "AudioIPC Server RPC",
+        move || {
+            audioipc::server_platform_init();
+            Ok(())
```

Comment:
nits: It seems we can change the second parameter of `core::spawn_thread` from `F: FnOnce() -> io::Result<()> + ..` to `F: FnOnce() -> () + ..` since the returned `Result` from `f` will be silently consumed anyway. By doing so, we no longer need to add an `Ok(())` at the end here. But it's another issue.

Reply:
Yeah, this is more a side effect of weird error handling with old (0.1) style futures, I think we can return useful errors here once we're up-to-date with the modern futures API.